src/hotspot/share/gc/z/zDriver.cpp

```diff
@@ -310,8 +310,7 @@ static bool should_preclean_young(GCCause::Cause cause) {
   // catch that above and perform young gen pre-cleaning.
   assert(!should_clear_soft_references(cause), "Clearing soft references without pre-cleaning young gen");
 
-  // Preclean young if implied by configuration
-  return ScavengeBeforeFullGC;
+  return false;
```

Comment:
I'm not sure if this is correct. We previously returned true here if the user didn't set ScavengeBeforeFullGC, right? Maybe we need a ZGC-specific flag to allow this "precleaning" to be turned off?

Reply:
Ahh, I missed that. Nowadays we typically put that kind of code in the GC-specific places and not in the shared code.